Platon/backend/app/search_engine/views.py

```diff
@@ -308,7 +327,77 @@ def get(self):
+    def get(self):
+        form = UpcomingEventsSearchForm(request.args)
+        if form.validate():
+            search_query = form.search_query.data.lower()
+            # Remove the punctuation in the search query
+            search_query = SearchEngine.remove_punctuation(search_query)
+            # Tokenize the search query
+            tokens = search_query.split()
+            # Get stopwords form the API
+            stopwords = SearchEngine.get_stopwords()
+            # Remove stopwords from the list of tokens
+            tokens = list(set(tokens) - set(stopwords))
+            # Get the semantically related list for our tokens
+            tokens = SearchEngine.semantic_related_list(tokens)
+            # List of user dictionaries that will be send as a search output
+            result_list = []
+            # Score of each user record
+            result_id_score = []
+            for token, score in tokens:
+                try:
+                    query = '(LOWER(id) REGEXP ".*{0}.*" OR LOWER(title) REGEXP ".*{0}.*" \
+                    OR LOWER(acronym) REGEXP ".*{0}.*" OR LOWER(location) REGEXP ".*{0}.*") \
+                    OR LOWER(date) REGEXP ".*{0}.*" OR LOWER(deadline) REGEXP ".*{0}.*" OR LOWER(link) REGEXP ".*{0}.*"'  \
+                    .format(token)
+                    sql_statement = "SELECT id,title,acronym,location,date,deadline,link FROM upcoming_events WHERE {}".format(query)
+                    result = db.engine.execute(sql_statement)
+                    for upcoming_event in result:
+                        id_list = [i[0] for i in result_id_score]
+                        if upcoming_event[0] not in id_list:
+                            if upcoming_event[-1] == 0:
+                                continue
+                            result_list.append({"id": upcoming_event[0], "title": upcoming_event[1], "acronym": upcoming_event[2],
+                                                "location": upcoming_event[3], "date": upcoming_event[4],
+                                                "deadline": upcoming_event[5], "link": upcoming_event[6]})
+                            result_id_score.append((upcoming_event[0], score))
+                        else:
+                            id_index = id_list.index(upcoming_event[0])
+                            result_id_score[id_index] = (upcoming_event[0], result_id_score[id_index][1] + score)
+                except:
+                    return make_response(jsonify({"error": "Database Connection Problem."}), 500)
+
+                # Sort result ids according to their scores
+                sorted_id_list = SearchEngine.sort_ids(result_id_score)
+                sorted_result_list = []
+                for upcoming_event_id, score in sorted_id_list:
+                    index = [upcoming_event["id"] for upcoming_event in result_list].index(upcoming_event_id)
+                    sorted_result_list.append(result_list[index])
```

Comment:
1.1.6.4 Users and guests should be able to sort search results of upcoming events according to the sorting criteria as date, alphabetical order.

According to this requirement, results shall be able to sort according to their dates and alphabetical order. Can you add such a feature?

Reply:
Added them too.